### ISSUE TITLE
[d3] Parallel dependent bug when integrating a dot product

### ISSUE THREAD

**Issue by whitney-powers** [DedalusProject/dedalus#152]
Hello Dedalus Team, 
I've found a parallel-dependent error on the tip of d3 with integrating a dot product. Here is a minimal working example. In this code, u is filled with ones on a 2d Cartesian domain. I define the ez unit vector, and take dot(u,ez) which should be a scalar array filled with ones. This is done correctly. We next try to integrate dot(u,ez) over the x axis. This should give us a 1D array filled with ones. In serial we successfully get this result, but in parallel only the rank 0 processor gives us this result. The remaining processors give us arrays filled with zeros.
Run this script like this to reproduce this bug. Strip the .txt from the file in order to run it. Github does not let me upload .py files directly
`mpiexec -n 4 python3 dot_integration.py`
[dot_integration.py.txt](https://github.com/DedalusProject/dedalus/files/7188811/dot_integration.py.txt)

**Comment by kburns** [comment]
It looks like this is a bug in the parallel evaluation of cartesian operators that affects many different operators. This has been addressed in #161, where the test script here now passes. For now let's keep this issue open as the tracking issue on this problem until that PR is merged.

**Comment by kburns** [comment]
#161 has now been merged into the `d3` branch. Can we double-check if these changes have fixed the issues in your other scripts?
